app/controllers/course_user_data_controller.rb

```diff
@@ -106,7 +106,8 @@ def update
         if @editCUD.save
           redirect_to(action: :show) && return
         else
-          redirect_to(action: :edit) && return
+          @editCUD.tweak ||= Tweak.new
+          render :edit
```

Comment:
Do you thin it is better to do `@cud.errors.full_messages` like we did in application controller?